graph/driver/driver.go

```diff
@@ -4,8 +4,9 @@ import (
 	"context"
 	"errors"
 
-	"github.com/ONSdigital/dp-code-list-api/models"
+	codelistModels "github.com/ONSdigital/dp-code-list-api/models"
```

Comment:
This feels like an odd cyclic dependency situation.  Have we considered moving the models into this package so that the apis only depend on this package rather than the opposite?

Possibly we are just leaving this until after the switch to using this implementation to make the transition simpler?

Reply:
thats definitely been playing on my mind. luckily the way things are abstracted, the models packages never really import much of anything so its (relatively) safe from cyclic issues, but it certainly feels messy.

it doesn't feel like the models should live in dp-graph, as they are generally used by the APIs to return data as well, so could be useful to other clients etc (and would like to avoid using this repo as a new go-ns like dumping ground where possible)


previously we've chosen against having one centralized 'models' repo for the APIs, but it might be worth revisiting. probably worth a chat early next week?

Reply:
a quick chat about this suggested that:
- its definitely a conversation we should be having, and there are issues with our current approch to models across the board
- we don't have a common understanding of what should and shouldnt be in a models package, or how much logic is appropriate for them
- we don't have a clear understandng of the bounds or amount of work involved in refactoring models to a common location

I'd suggest this is a conversation to have before the team move on to CMD work, but that it's probably signficant enough work that it will have to be done outside of our current neo refactoring efforts :)